relayer/processor/path_processor_internal.go

```diff
@@ -369,8 +369,81 @@ func (pp *PathProcessor) updateClientTrustedState(src *pathEndRuntime, dst *path
 	}
 }
 
+func (pp *PathProcessor) appendInitialMessageIfNecessary(msg MessageLifecycle, pathEnd1Messages, pathEnd2Messages *pathEndMessages) {
+	if msg == nil || pp.sentInitialMsg {
+		return
+	}
+	switch m := msg.(type) {
+	case *PacketMessageLifecycle:
+		if m.Initial == nil {
+			break
+		}
+		channelKey, err := PacketInfoChannelKey(m.Initial.Action, m.Initial.Info)
+		if err != nil {
+			pp.log.Error("Unexpected error checking packet message",
+				zap.String("action", m.Termination.Action),
+				zap.Any("channel", channelKey),
+				zap.Error(err),
+			)
+			break
+		}
+		if !pp.IsRelayedChannel(m.Initial.ChainID, channelKey) {
+			break
+		}
+		if m.Initial.ChainID == pp.pathEnd1.info.ChainID {
+			pathEnd1Messages.packetMessages = append(pathEnd1Messages.packetMessages, packetIBCMessage{
+				action: m.Initial.Action,
+				info:   m.Initial.Info,
+			})
+		} else if m.Initial.ChainID == pp.pathEnd2.info.ChainID {
+			pathEnd2Messages.packetMessages = append(pathEnd2Messages.packetMessages, packetIBCMessage{
+				action: m.Initial.Action,
+				info:   m.Initial.Info,
+			})
+		}
+	case *ConnectionMessageLifecycle:
+		if m.Initial == nil {
+			break
+		}
+		if !pp.IsRelevantConnection(m.Initial.ChainID, m.Initial.Info.ConnID) {
+			break
+		}
+		if m.Initial.ChainID == pp.pathEnd1.info.ChainID {
+			pathEnd1Messages.connectionMessages = append(pathEnd1Messages.connectionMessages, connectionIBCMessage{
+				action: m.Initial.Action,
+				info:   m.Initial.Info,
+			})
+		} else if m.Initial.ChainID == pp.pathEnd2.info.ChainID {
+			pathEnd2Messages.connectionMessages = append(pathEnd2Messages.connectionMessages, connectionIBCMessage{
+				action: m.Initial.Action,
+				info:   m.Initial.Info,
+			})
+		}
+	case *ChannelMessageLifecycle:
+		if m.Initial == nil {
+			break
+		}
+		if !pp.IsRelevantChannel(m.Initial.ChainID, m.Initial.Info.ChannelID) {
+			break
+		}
+		if m.Initial.ChainID == pp.pathEnd1.info.ChainID {
+			pathEnd1Messages.channelMessages = append(pathEnd1Messages.channelMessages, channelIBCMessage{
+				action: m.Initial.Action,
+				info:   m.Initial.Info,
+			})
+		} else if m.Initial.ChainID == pp.pathEnd2.info.ChainID {
+			pathEnd2Messages.channelMessages = append(pathEnd2Messages.channelMessages, channelIBCMessage{
+				action: m.Initial.Action,
+				info:   m.Initial.Info,
+			})
+		}
+	}
+
+	pp.sentInitialMsg = true
```

Comment:
This final statement is why the `switch` above has many `break` statements instead of `return` statements. Reading through a long switch with a lot of `break`s tells the reader there is something important after the switch.

Could this assignment happen immediately before the switch so that the `break` statements could be `return`s instead?